src/classes/Ingredient.js

```diff
@@ -1,14 +1,26 @@
 class Ingredient {
   constructor(id, quantity, ingredientsData) {
-    this.id = id;
-    this.name = ingredientsData.find(ingredient => this.id === ingredient.id).name;
-    this.estimatedCostInCents = ingredientsData.find(ingredient => this.id === ingredient.id).estimatedCostInCents;
+    this.id = id || 0;
+    this.name = this.findIngredientName(ingredientsData);
+    this.estimatedCostInCents = this.findEstimatedCostInCents(ingredientsData);
```

Comment:
Adding some default values and moving iterators into thier own functions instead of using them directly in the constructor